src/github/client/github-client.ts

```diff
@@ -16,19 +16,19 @@ export class GitHubClient {
 
 	constructor(
 		logger: Logger = getLogger("gitHub-client"),
-		baseUrl?: string
+		hostname?: string
 	) {
 		this.logger = logger;
 
-		// baseUrl is undefined when FF is false
+		// hostname is undefined when FF is false
 		// if FF is true and the githubAppId field is empty, it is set to https://api.github.com
 		// TODO - clean this logic up once we remove the GHE_SERVER flag
-		if (baseUrl == undefined || baseUrl === "https://api.github.com") {
+		if (hostname == undefined || hostname === GITHUB_CLOUD_HOSTNAME) {
```

Comment:
The logic is changed here, right? Do you mind adding some tests please?

---

And can you please also explain why the logic in current production code doesn't work `for cloud`?
Is it, in prod as for now, the `hostname` will ALWAYS be `undefined` in `cloud`?

Reply:
This was done as a result of changes done in the `get-github-client-config`, cause the `baseUrl` was incorrect in `getGitHubApiUrl`.
Reverted these changes to avoid all these confusion.